### PR TITLE
Put example notebooks in sub-directories, run `nb-clean`, and make minor edits

### PR DESCRIPTION
**Pull request by namurphy**
- I moved the notebooks in `docs/notebooks` to the appropriate subdirectories.  Most were associated with `plasmapy.formulary`.
 - I ran nb-clean on all of the notebooks that were not pre-executed.  This removes some unnecessary metadata.
 - I renamed `physics.ipynb` to `iter.ipynb`, since that reflected the obsolete `plasmapy.physics` subpackage which was renamed to `plasmapy.formulary`.
 - I made a few minor edits to some notebooks, mostly for links & formatting.

### CODE REVIEW

File: docs/formulary/lengths.rst [hunk 1]
```diff
@@ -8,6 +8,9 @@ Length Plasma Parameters (`plasmapy.formulary.lengths`)
 
 .. automodapi:: plasmapy.formulary.lengths
 
+Examples
+========
+
 .. nbgallery::
```
Comment (rocco8773) [inline docs/formulary/lengths.rst:14]: You would add the `magnetosphere` notebook here.

File: docs/formulary/lengths.rst [hunk 2]
```diff
@@ -4,4 +4,15 @@
...
 .. automodapi:: plasmapy.formulary.lengths
+
+Examples
+========
+
+.. nbgallery::
+    :caption: Examples
+
+    /notebooks/formulary/ExB_drift
+    /notebooks/formulary/iter
```
Comment (rocco8773) [inline docs/formulary/lengths.rst:18]: You would add the `magnetosphere` notebook here.

```suggestion
    /notebooks/formulary/ExB_drift
    /notebooks/formulary/iter
    /notebooks/formulary/magnetosphere
```

Comment (namurphy) [reply docs/formulary/lengths.rst:18]: Good catch!  I must have been spacing out on that one.

File: docs/notebooks/formulary/iter.ipynb [hunk 3]
```diff
@@ -73,9 +64,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Note that we can also neglect the unit for the concentration, as\n",
-    "1/m^3 is the a standard unit for this kind of Quantity:\n",
-    "\n"
+    "[Quantity]: https://docs.astropy.org/en/stable/api/astropy.units.Quantity.html#astropy.units.Quantity\n",
+    "\n",
+    "Note that we can also neglect the unit for the concentration, as m$^{-3}$ is the a standard unit for this kind of Quantity:"
```
Comment (rocco8773) [inline docs/notebooks/formulary/iter.ipynb:69]:
```suggestion
    "Note that we can also neglect the unit for the concentration, as m$^{-3}$ is the a standard unit for this kind of [Quantity]:"
```

**Review by rocco8773 — APPROVED** [review]

**Pull request merged**